pkg/restore/restore.go

```diff
@@ -1249,13 +1249,36 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
-		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
-		if err != nil {
-			ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
-			warnings.Add(namespace, err)
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
+	if err != nil {
+		ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
+		warnings.Add(namespace, err)
+		return warnings, errs
+	}
+
+	if !isAlreadyExistsError {
+		if !apierrors.IsNotFound(err) {
+			objectExists = true
+		}
+
+		if err != nil && objectExists {
+			errs.Add(namespace, err)
 			return warnings, errs
 		}
+	}
+
+	if isAlreadyExistsError || objectExists {
+		if fromCluster != nil {
```

Comment:
This should be "if fromCluster == nil"

Reply:
If
	fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
is nil the first time.. what make this second call any different?

Reply:
So we should only call Get earlier if it's *not* an already exists error (and in that case, we ignore the error if we can't retrieve it and instead add the error from the create call to the restore). This call will be run for the already exists error case.